### PR TITLE
chore(gpu): extend integer multiplication to N other than 2048

### PR DESCRIPTION
**Pull request by agnesLeroy**
closes:  https://github.com/zama-ai/tfhe-rs-internal/issues/470

### Check-list:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by bbarbakadze** [comment]
can't run tests on my laptop, will have a look

**Review by bbarbakadze — APPROVED** [review]

**Comment by pdroalves** [comment]
FYI, yesterday I've fixed the invalid memory access that happened when the PBS runs on `NOSM` mode. However, I found out that there was also a hidden bug when `pbs_level > 1`. I'm on it, but that's why it's taking so long to fix this PR.

**Comment by pdroalves** [comment]
@agnesLeroy Looks like the last issue is an out-of-memory error. I guess that is not a surprise considering that the `lwe_chunk_size` is fixed and `polynomial_size = 8192` consumes a lot of memory. Maybe we can first merge `chore/gpu/lwechunksize_refactor` first and then run again tests. Wdyt?

**Comment by agnesLeroy** [comment]
Getting an out of memory error in the tests :disappointed:

**Comment by pdroalves** [comment]
@slab-ci integer_gpu_full_bench

File: backends/tfhe-cuda-backend/cuda/src/device.cu [hunk 1]
```diff
@@ -142,6 +142,8 @@ __global__ void cuda_set_value_kernel(Torus *array, Torus value, Torus n) {
 template <typename Torus>
 void cuda_set_value_async(cudaStream_t *stream, Torus *d_array, Torus value,
                           Torus n) {
+  if (n == 0)
+    return;
```
Comment (agnesLeroy) [inline backends/tfhe-cuda-backend/cuda/src/device.cu:146]: I don't think we need this one

Comment (pdroalves) [reply backends/tfhe-cuda-backend/cuda/src/device.cu]: Removed.

**Comment by pdroalves** [comment]
@slab-ci integer_gpu_full_bench

**Comment by agnesLeroy** [comment]
rtx4090 benchmarks on integers are green, I think we're good to merge :slightly_smiling_face:

**Pull request merged**